VirtualList/tests/VirtualGridList-specs.js

```diff
@@ -0,0 +1,353 @@
+		const headingZeros = Array(itemNumberDigits).join('0');
+
+		for (let i = 0; i < dataSize; i++) {
+			const count = (headingZeros + i).slice(-itemNumberDigits),
+				text = `Item ${count}`,
+				subText = `SubItem ${count}`,
+				color = Math.floor(Math.random() * (0x1000000 - 0x101010) + 0x101010).toString(16),
+				source = svgGenerator(300, 300, color, 'ffffff', `Image ${i}`);
+
+			items.push({text, subText, source});
+		}
+	});
+
+	afterEach(() => {
+		clientSize = null;
+		dataSize = null;
+		items = null;
+		itemSize = null;
+		renderItem = null;
+	});
+
+	test('should render a list of \'items\'', () => {
+		render(
+			<VirtualGridList
+				clientSize={clientSize}
+				dataSize={dataSize}
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+			/>
+		);
+
+		const expected = 'Item 000SubItem 000';
+		const actual = screen.getByRole('list').children.item(0).textContent;
+
+		expect(actual).toBe(expected);
+	});
+
+	test('should not render items when clientSize is not given', () => {
+		render(
+			<VirtualGridList
+				dataSize={dataSize}
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+			/>
+		);
+
+		const expected = 0;
+		const actual = screen.getByRole('list').children.length;
+
+		expect(actual).toBe(expected);
+	});
+
+	test('should render (clientHeight / itemHeight + overhang) items', () => {
+		render(
+			<VirtualGridList
+				clientSize={clientSize}
+				dataSize={dataSize}
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+			/>
+		);
+
+		const expected = 24; // (4 * (720 / 240)) + 4 * 3
+		const actual = screen.getByRole('list').children.length;
+
+		expect(actual).toBe(expected);
+	});
+
+	test('should re-render (clientHeight / itemHeight + overhang) items after changing client size', () => {
+		const {rerender} = render(
+			<VirtualGridList
+				clientSize={clientSize}
+				dataSize={dataSize}
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+			/>
+		);
+
+		const newClientSize = {clientWidth: 1280, clientHeight: 360};
+
+		rerender(
+			<VirtualGridList
+				clientSize={newClientSize}
+				dataSize={dataSize}
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+			/>
+		);
+
+		const expected = 20; // (4 * (360 / 240)) + 4 * 3
+		const actual = screen.getByRole('list').children.length;
+
+		expect(actual).toBe(expected);
+	});
+
+	test('should render only one scrollbar', () => {
+		render(
+			<VirtualGridList
+				clientSize={clientSize}
+				dataSize={dataSize}
+				direction="horizontal"
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+			/>
+		);
+
+		const expected = 2; // One for the list and another for the horizontal scrollbar
+		const actual = screen.getByRole('list').parentElement.parentElement.children.length;
+
+		expect(actual).toBe(expected);
+	});
+
+	describe('Scrollbar visibility', () => {
+		test(
+			'should render both horizontal and vertical scrollbars when \'horizontalScrollbar\' and \'verticalScrollbar\' are "visible"',
```

Comment:
If you are going to put `\'`  to props, please put it for all props in the test description.